In systemd I have seen this error, using it as a service:

vlock-start[14567]: vlock-new: could not activate new terminal: Interrupted system call

I think this should fix that.

Also on github: https://github.com/viric/vlock/commit/781a26087f83c7247601b6f82f784cca9266694e

diff --git a/modules/new.c b/modules/new.c
index e9b15fb..7aed640 100644
--- a/modules/new.c
+++ b/modules/new.c
@@ -103,9 +103,19 @@ static char *get_console_name(int n)
  * file descriptor. */
 static int activate_console(int consfd, int vtno)
 {
-  int c = ioctl(consfd, VT_ACTIVATE, vtno);
+  int c;
+  do {
+    c = ioctl(consfd, VT_ACTIVATE, vtno);
+  } while(c != 0 && errno == EINTR);
 
-  return c < 0 ? c : ioctl(consfd, VT_WAITACTIVE, vtno);
+  if (c < 0)
+      return c;
+
+  do {
+    c = ioctl(consfd, VT_WAITACTIVE, vtno);
+  } while(c != 0 && errno == EINTR);
+
+  return c;
 }
 
 struct new_console_context {
